From 1e4909abf2b4dbbc093b4a4ffdf870457b65a5fb Mon Sep 17 00:00:00 2001
From: Kevin McCarthy <kevin@8t8.us>
Date: Mon, 9 May 2016 14:06:59 -0700
Subject: [PATCH 31/31] Back out wrapper sys_socket.h workaround (23334e967dd7)

After renaming all internal macros from M_* to MUTT_*, the
sys_socket.h workaround is no longer necessary.
---
 Makefile.am   |  2 +-
 getdomain.c   |  2 +-
 mutt_sasl.c   |  2 +-
 mutt_socket.c |  2 +-
 mutt_tunnel.c |  2 +-
 sys_socket.h  | 51 ---------------------------------------------------
 6 files changed, 5 insertions(+), 56 deletions(-)
 delete mode 100644 sys_socket.h

diff --git a/Makefile.am b/Makefile.am
index c32440b..b39c053 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -74,7 +74,7 @@ EXTRA_DIST = COPYRIGHT GPL OPS OPS.PGP OPS.CRYPT OPS.SMIME TODO UPDATING \
 	README.SSL smime.h group.h \
 	muttbug pgppacket.h depcomp ascii.h BEWARE PATCHES patchlist.sh \
 	ChangeLog mkchangelog.sh mutt_idna.h \
-	snprintf.c regex.c crypt-gpgme.h hcachever.sh.in sys_socket.h \
+	snprintf.c regex.c crypt-gpgme.h hcachever.sh.in \
 	txt2c.c txt2c.sh version.sh check_sec.sh version.h
 
 EXTRA_SCRIPTS = smime_keys
diff --git a/getdomain.c b/getdomain.c
index 9bc0c91..209848b 100644
--- a/getdomain.c
+++ b/getdomain.c
@@ -24,7 +24,7 @@
 #include <unistd.h>
 #include <netdb.h>
 #include <sys/types.h>
-#include "sys_socket.h"
+#include <sys/socket.h>
 
 #include "mutt.h"
 
diff --git a/mutt_sasl.c b/mutt_sasl.c
index ffea2a6..99dc22a 100644
--- a/mutt_sasl.c
+++ b/mutt_sasl.c
@@ -30,7 +30,7 @@
 #include <errno.h>
 #include <netdb.h>
 #include <sasl/sasl.h>
-#include "sys_socket.h"
+#include <sys/socket.h>
 #include <netinet/in.h>
 
 static int getnameinfo_err(int ret)
diff --git a/mutt_socket.c b/mutt_socket.c
index 0499ba1..a2c489f 100644
--- a/mutt_socket.c
+++ b/mutt_socket.c
@@ -40,7 +40,7 @@
 #ifdef HAVE_SYS_TIME_H
 #include <sys/time.h>
 #endif
-#include "sys_socket.h"
+#include <sys/socket.h>
 #ifdef HAVE_SYS_SELECT_H
 #include <sys/select.h>
 #endif
diff --git a/mutt_tunnel.c b/mutt_tunnel.c
index 4917216..b43bbf9 100644
--- a/mutt_tunnel.c
+++ b/mutt_tunnel.c
@@ -27,7 +27,7 @@
 
 #include <netinet/in.h>
 #include <sys/types.h>
-#include "sys_socket.h"
+#include <sys/socket.h>
 #include <sys/wait.h>
 #include <fcntl.h>
 #include <errno.h>
diff --git a/sys_socket.h b/sys_socket.h
deleted file mode 100644
index eb56c14..0000000
--- a/sys_socket.h
+++ /dev/null
@@ -1,51 +0,0 @@
-/*
- * Copyright (C) 2016 Kevin J. McCarthy <kevin@8t8.us>
- *
- *     This program is free software; you can redistribute it and/or modify
- *     it under the terms of the GNU General Public License as published by
- *     the Free Software Foundation; either version 2 of the License, or
- *     (at your option) any later version.
- *
- *     This program is distributed in the hope that it will be useful,
- *     but WITHOUT ANY WARRANTY; without even the implied warranty of
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *     GNU General Public License for more details.
- *
- *     You should have received a copy of the GNU General Public License
- *     along with this program; if not, write to the Free Software
- *     Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
- */
-
-/* Solaris, OpenIndiana, and probably other derivatives
- * are including sys/stream.h inside their sys/socket.h.
- *
- * This include file is defining macros MUTT_CMD and MUTT_READ which
- * are conflicting with the same macros Mutt defines in mutt.h
- *
- * To minimize breakage with out-of-tree patches, this is a workaround.
- */
-
-#ifdef MUTT_CMD
-# define MUTT_ORIG_CMD MUTT_CMD
-# undef MUTT_CMD
-#endif
-
-#ifdef MUTT_READ
-# define MUTT_ORIG_READ MUTT_READ
-# undef MUTT_READ
-#endif
-
-#include <sys/socket.h>
-
-#undef MUTT_CMD
-#undef MUTT_READ
-
-#ifdef MUTT_ORIG_CMD
-# define MUTT_CMD MUTT_ORIG_CMD
-# undef MUTT_ORIG_CMD
-#endif
-
-#ifdef MUTT_ORIG_READ
-# define MUTT_READ MUTT_ORIG_READ
-# undef MUTT_ORIG_READ
-#endif
-- 
2.8.2

